### ISSUE TITLE
Stop sending updates after x time of idling

### ISSUE THREAD

**Issue by mercxry** [andweeb/presence.nvim#40]
Hello, would it be possible to add a setting to stop sending updates to discord presence after x seconds/minutes of idling?

Thank you for doing my previous request so quickly 😁

**Comment by andweeb** [comment]
Hi @mercxry! If I'm understanding correctly, what you're suggesting is a configuration option to expire/cancel a file's Rich Presence after some timeout, right?

I think that's a good idea, I have seen some weird looking Rich Presence activity when putting my computer to sleep overnight, like 10:12:47 elapsed (which basically says I've been editing 10 hours... 😅). It might make more sense to have it as the default behavior actually, and allow users to configure the timeout value.

Is this what you were suggesting?

**Comment by mercxry** [comment]
Yes, a VSCode extension that I use has the "Idling" state, but if it's not too much I would like to have a way to cancel the state (so it doesn't show nvim at all in the activity tab) until I start typing again.

I have the same problem, I rarely shut down my computer and I always keep nvim open in my terminal so I get high editing times like you.

**Comment by ModProg** [comment]
I would also like to have something like this.

If you are set on a way this should be implemented, I would also be interested in implementing it.

I would like to have a timeout value that just says after Neovim was not in focus for 5 minutes stop showing the presence (this also means the automatic game presence would kick in) but when Neovim is refocused it should start with the time before it paused.

So that if you edit for 2h, take a 40-min break, and then continue to work it would start at 2h (2:05h technically due to timeout time).

**Comment by ModProg** [comment]
Alright, thanks for the extensive explanaition, I'll have a look on the weekend.

Is there a reason not to use `FocusLost` events for the cancel behaviour?
![grafik](https://user-images.githubusercontent.com/11978847/179683505-ce65a227-f251-4000-a6f1-8e77ede2c574.png)

**Comment by andweeb** [comment]
No pressure! :)

> Is there a reason not to use `FocusLost` events for the cancel behaviour?

Good question. The `FocusLost` event would work for cases in which Neovim no longer has active focus (e.g. user minimizes the nvim window), but it wouldn't cover the case in which the user _is_ focused on nvim and idling, specifically the scenarios that were discussed in the original comments:
* User edits a file on nvim but goes afk
* User edits a file on nvim but puts their laptop to sleep

That's why I started with using `CursorMoved`, as I'm not sure other events would fully capture what we need in order to start the idle timer. Let me know if you find a potentially better option, since the `CursorMoved` event has a pretty big caveat that it will be called very frequently, as noted in the helptext:
> Careful: This is triggered very often, don't do anything that the user does not expect or that is slow.

**Comment by ModProg** [comment]
Maybe this can be a setting, as there is also the problem that some tiling terminals don't send FocusLost events.

**Comment by oriongonza** [comment]
It doesn't need to be terribly accurate. I have something like 6 instances of nvim open at any time across many tmux sessions. I just want that if I'm not working on one for like 10m it stops showing it, so an autocmd like BufWrite and BufEnter would be okay. If you're not saving your work or moving between files you're probably not working that much.
